car_information/car_information_service.py

```diff
@@ -20,11 +20,39 @@ def identify_license_plate(path, image):
     try:
         r = requests.post(OPEN_ALPR_ENDPOINT, data=img_base64)
         result_json = r.json()
+        results = result_json["results"]
 
-        return result_json["results"][0]["plate"]
-    except:
-        return None
+        if len(results) == 0:
+            raise LicensePlateNotDetectedException("Could not detect any license plates")
 
+        license_plate = get_largest_plate(results)
+
+        if not is_valid_license_plate(license_plate):
+            raise IllegalLicensePlateException("The license plate does not follow Norwegian convention")
+
+        return license_plate
+    finally:
+        pass
+
+
+def get_largest_plate(result_json):  # If several plates are in the picture, choose the on that takes the most space
+    largest_plate = [0, 0]  # [0] is the plate index, [1] is the square size of the given plate
+    for i in range(len(result_json)):
+        plate_height = result_json[i].get("vehicle_region").get("height")
+        plate_width = result_json[i].get("vehicle_region").get("width")
```

Comment:
What about making a new variable, e.g. `vehicle_region = result_json[i].get("vehicle_region")`? This is so that you don't fetch the result element reduntantly, like you're doing in these two lines.

Reply:
Done.